druid/examples/cursor.rs

```diff
@@ -13,44 +13,28 @@
 // limitations under the License.
 
 //! An example showing how to change the mouse cursor.
+//! Clicking the button should switch your cursor, and
+//! the last cursor should be a custom image. Custom
+//! image cursors cannot be created before the window is
+//! open so we have to work around that. When we receive the
+//! `WindowConnected` command we initiate the cursor.
 
 use druid::{
-    AppLauncher, Color, Cursor, CursorDesc, Data, Env, ImageBuf, Lens, LocalizedString, Widget,
-    WidgetExt, WindowDesc,
+    AppLauncher, Color, Cursor, CursorDesc, Data, Env, ImageBuf, Lens, LocalizedString, WidgetExt,
+    WindowDesc,
 };
 
 use druid::widget::prelude::*;
 use druid::widget::{Button, Controller};
 
-use std::sync::Arc;
+use std::rc::Rc;
 
-#[derive(Clone, Data, Lens)]
```

Comment:
These patches are making me think more generally about how I would structure these files, in terms of where to put `main()`, where to put `AppState`, etc.

As a *general* rule, when I write rust, I put all of the public types at the top of the file, with their implementations below. I think this makes it easier to get a general sense of what a file will contain by looking at the start of the file. I don't always follow this exactly; especially for things like iterator types or other less important things, I will sometimes put them elsewhere in a file.

If i had to pick a rule for `main()` I would say it should always just go at the end of the example. 🤷

Reply:
I agree with you for a normal druid application, but examples usually show something off, I have tried to put whatever is being shown off at the top and leave the less important things at the bottom.

With examples people always copy paste of course, but I think file layout is something people will always change to fit their liking, so I didn't prioritise the notnal layout I would use for structuring the program (or what was already there).

Maybe I am thinking too much about this, maybe I should just structure them like a normal program, but my experience is that examples that do it that way don't help me. These are short examples so they won't live or die by it, but I still want to make them to the point.

Reply:
I think your reasoning makes sense!